airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_read_processor.py

```diff
@@ -148,9 +148,14 @@ def on_exception(self, exception: StreamThreadException) -> Iterable[AirbyteMess
         """
         self._flag_exception(exception.stream_name, exception.exception)
         self._logger.exception(f"Exception while syncing stream {exception.stream_name}", exc_info=exception.exception)
-        yield AirbyteTracedException.from_exception(
-            exception, stream_descriptor=StreamDescriptor(name=exception.stream_name)
-        ).as_airbyte_message()
+
+        stream_descriptor = StreamDescriptor(name=exception.stream_name)
+        if isinstance(exception.exception, AirbyteTracedException):
+            yield exception.exception.as_airbyte_message(stream_descriptor=stream_descriptor)
```

Comment:
I'm not sure about how I feel by passing `stream_descriptor` as part of `as_airbyte_message`. On one hand, we mentioned it was probably better to deprecate. On the other, I'm not sure all AirbyteTracedException have the context of the stream and hence might benefit from the added information